tardis/utilities/utils.py

```diff
@@ -42,3 +56,26 @@ def htcondor_csv_parser(htcondor_input, fieldnames, delimiter="\t", replacements
                 key: value if value not in replacements.keys() else replacements[value]
                 for key, value in row.items()
             }
+
+
+def slurm_cmd_option_formatter(options: AttributeDict) -> str:
+    option_prefix = dict(short="-", long="--")
+    option_separator = dict(short=" ", long="=")
+
+    option_string = ""
+
+    for option_type in ("short", "long"):
+        try:
+            tmp_option_string = cmd_option_formatter(
+                getattr(options, option_type),
+                prefix=option_prefix[option_type],
+                separator=option_separator[option_type],
+            )
+        except AttributeError:
+            pass
+        else:
+            if option_string:  # add additional space between short and long options
+                option_string += " "
+            option_string += tmp_option_string
+
+    return option_string
```

Comment:
Consider not to join the ``option_string`` one by one but instead creating a list and joining with whitespace when returning the result.
Further, I think we should return an error in case an option cannot properly be formatted/interpreted. This might result in unexpected behaviour for the user, which we don't want.

Reply:
Hmm, I think this is a misunderstanding. We have to join short and long options separately. Since we do not know if short options is present, we have to use the `if` conditional to add a separating white space in case they are.  Options are optional, so if no options are present the occuring `AttributeError` should be ignored. Do you want me to add a log output in that case? It is not an error in any case.

Reply:
Just to clarify: since ``options`` is an ``AttributeDict``, it will raise ``AttributeError`` instead of ``KeyError`` when a key is missing? So the ``except AttributeError`` path basically handles the case that the configuration does not include a ``short`` and/or ``long`` key?

Reply:
In terms of maintainability this is not great, but I think we can go for it. 
@giffels, I would approve if you don't want to change anything else.

Reply:
Exactly!

Reply:
@eileen-kuehn ticket #152 has been opened on configuration management.